### ISSUE TITLE
CORS warning using Picture

### ISSUE THREAD

**Issue by mlomb** [amcharts/amcharts5#822]
**Bug description**

When you use an image from a third-party domain like so:
```js
Picture.new(root, {
    src: "https://www.amcharts.com/wp-content/uploads/assets/logos/ford.png",
    centerX: p50,
    centerY: p50,
    width: percent(30),
    isMeasured: true,
})
```

 the following warning is displayed in the console:

> Image "\<image URL>" is loaded from different host and is not covered by CORS policy. For more information about the implications read here: https://www.amcharts.com/docs/v5/concepts/cors

But in this case that image has the header "access-control-allow-origin: *" and **it renders as expected**.
Also the provided link is broken.

**Reproducible example**

Open the JSFiddle or [Codepen](https://codepen.io/pen?&prefill_data_id=310aaac1-f35f-4a17-8487-99859f81b27c) of the [Multilevel Tree Map](https://www.amcharts.com/demos/multilevel-tree-map/) demo and open the console

**Comment by Pauan** [comment]
In order to enable CORS, the server must respond properly to the [preflight request](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request), sending `Access-Control-Allow-Origin: *` by itself isn't enough. It must also handle authentication correctly:

https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

amCharts just draws the image into the canvas, we can't control the [browser's security CORS restrictions](https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image#security_and_tainted_canvases).

**Comment by mlomb** [comment]
Are the image requests [simple requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests)? Which don't trigger the prefilight request but require the "Access-Control-Allow-Origin" for the client to be able to read the response.

I did not know that canvas can be "tinted", I tried using `toDataURL` in the chart canvas and got:
> Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.

Which I think its because the images were not loaded using CORS with preflight.

The problem I have is that the warning is flooding my console. Requests are not failing since we are able to see the images correctly in the chart, so I think the message can be delayed to when the canvas being tinted becomes a problem (for example when exporting as an image, which I don't need).

**Comment by Pauan** [comment]
Simple requests only apply to forms, not canvas images. Images aren't only used for export, they are also used for hit detection (e.g. if you add a `click` event listener to the Picture). So you really shouldn't be using tainted images, all of the images should be CORS-compatible.

**Comment by mlomb** [comment]
I think we don't need full CORS with preflight for this request since it is just a GET with no additional headers. The server responds with **Access-Control-Allow-Origin: \*** which is very common in CDNs (no need to do the preflight rountrip) so I still think it should classify as a simple request (from what the docs says and my experience).

I investigated the code a bit and found where images are loaded:

https://github.com/amcharts/amcharts5/blob/1c392e494c88088573729eca2c58cdf8077254f1/src/.internal/core/render/Picture.ts#L128-L130

Setting the `crossOrigin` attribute to "anonymous" makes the request CORS aware and the image is no longer tinted! (and the warning goes away)

I think this could be configured in the Picture settings with a `crossOrigin` field or similar (alongside `src`) or maybe enable it by default, not sure why it was commented, it may have side effects I'm not aware.

**Comment by Pauan** [comment]
I've talked to the team, and we don't remember why we chose to not use `crossOrigin`.

We're going to experiment with enabling it by default, and also adding a new `cors` setting.

**Comment by martynasma** [comment]
Implemented in [5.3.6](https://github.com/amcharts/amcharts5/releases/tag/5.3.6).

## [5.3.6] - 2023-02-27

### Added
- New `cors` setting for `Picture` (defaults to `"anonymous"`).
- `beginPath` method added to `CanvasGraphics`.

### Fixed
- Theme states no longer override user states.
- In some specific cases with stacked axes series position could be shifted up.
- In some specific cases a hairline of a column of a `ColumnSeries` could remain visible on the plot area even if the whole column was out of visible axis bounds.
- A `ghostLabel` was causing some interfering issues with interactive labels of an axis.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
